### PR TITLE
set hostname from cloud-init

### PR DESCRIPTION
**Pull request by wlan0**
@ibuildthecloud

### CODE REVIEW

File: cmd/cloudinit/hostname/hostname.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+func SetHostname(hostname string) error {
+	osType := getOsType()
+
+	if osType == "busybox" {
+		return bb_setHostname(hostname)
+	}
+	return bb_setHostname(hostname)
+}
+
+func getOsType() string {
```
Comment (ibuildthecloud) [inline cmd/cloudinit/hostname/hostname.go]: Your already using this pattern in https://github.com/rancherio/os/pull/109/files.  Can we not repeat code somehow?

File: cmd/cloudinit/hostname/bb_hostname.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	}
+	f, err := os.OpenFile("/etc/hostname", os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	n, err := f.Write([]byte(hostname))
+	if err != nil {
+		return err
+	}
+	if n != len([]byte(hostname)) {
```
Comment (ibuildthecloud) [inline cmd/cloudinit/hostname/bb_hostname.go]: This isn't possible.  A short write will always result in an error.

File: cmd/cloudinit/hostname/bb_hostname.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	"errors"
+	"os"
+	"syscall"
+)
+
+func bb_setHostname(hostname string) error {
+	if err := syscall.Sethostname([]byte(hostname)); err != nil {
+		return err
+	}
+	f, err := os.OpenFile("/etc/hostname", os.O_WRONLY|os.O_TRUNC, 0644)
```
Comment (ibuildthecloud) [inline cmd/cloudinit/hostname/bb_hostname.go]: Can this just be simply don't with ioutil.WriteFile()?

File: cmd/cloudinit/cloudinit.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ import (
 	"github.com/coreos/coreos-cloudinit/initialize"
 	"github.com/coreos/coreos-cloudinit/pkg"
 	"github.com/coreos/coreos-cloudinit/system"
+	hostConf "github.com/rancherio/os/cmd/cloudinit/hostname"
```
Comment (ibuildthecloud) [inline cmd/cloudinit/cloudinit.go]: what's the conflict?  Why can't this just be the default package name?

File: cmd/cloudinit/cloudinit.go [hunk 5]
```diff
@@ -178,6 +179,13 @@ func Main() {
 		os.Exit(0)
 	}
 
+	if cc.Hostname != "" {
+		//set hostname
+		if err := hostConf.SetHostname(cc.Hostname); err != nil {
+			log.Fatalf("%v", err)
```
Comment (ibuildthecloud) [inline cmd/cloudinit/cloudinit.go]: It's this the same as `log.Fatal(err)`?

File: cmd/cloudinit/cloudinit.go (hunk 4 above)
Comment (wlan0) [inline cmd/cloudinit/cloudinit.go]: There is no conflict. Just calling hostname package as 'hostConf' everywhere

File: cmd/cloudinit/hostname/bb_hostname.go (hunk 3 above)
Comment (wlan0) [inline cmd/cloudinit/hostname/bb_hostname.go]: yep. Will make the change

File: cmd/cloudinit/cloudinit.go (hunk 5 above)
Comment (wlan0) [inline cmd/cloudinit/cloudinit.go]: yep

**Comment by wlan0** [comment]
@ibuildthecloud I think i've addressed all of your comments

File: util/util.go [hunk 6]
```diff
@@ -17,6 +18,22 @@ var (
 	letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
 )
 
+func GetOSType() string {
+	f, err := os.Open("/etc/os-release")
```
Comment (ibuildthecloud) [inline util/util.go]: defer close()

Comment (ibuildthecloud) [inline util/util.go]: or just use ReadFile

Comment (wlan0) [inline util/util.go]: fixed

**Comment by wlan0** [comment]
closes #100

**Pull request closed without merging**